packages/client/src/generation/generateClient.ts

```diff
@@ -604,11 +604,11 @@ async function copyRuntimeFiles({ from, to, runtimeName, sourceMaps }: CopyRunti
  * Attempts to delete the output directory.
  * @param finalOutputDir
  */
-async function deleteOutputDir(finalOutputDir: string) {
+async function deleteOutputDir(finalOutputDir: string, datamodel: string) {
   try {
     debug(`attempting to delete ${finalOutputDir} recursively`)
     // we want to make sure that if we delete, we delete the right directory
-    if (require(`${finalOutputDir}/package.json`).name === GENERATED_PACKAGE_NAME) {
+    if (require(`${finalOutputDir}/package.json`).name === getUniquePackageName(datamodel)) {
```

Comment:
The fix.

Reply:
Does not necessary needs changing right now, just food for thought. Would it make sense to change it to something like this?

```suggestion
    if (require(`${finalOutputDir}/package.json`).startsWith(GENERATED_PACKAGE_NAME)) {
```

Otherwise, if schema changed at the same time `--no-engine` flag was added to invocation, engine won't be deleted.

Reply:
Good catch, I'll add a test for that